persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -376,56 +321,82 @@ private static Relation computeRelation(String fieldName, Entity entity, Entity
         return relBuilder.build();
     }
 
-    public static SyntaxTree generateClientSyntaxTree(Module entityModule) throws BalException {
-        NodeList<ImportDeclarationNode> imports = AbstractNodeFactory.createEmptyNodeList();
-        NodeList<ModuleMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createEmptyNodeList();
-
-        MinutiaeList commentMinutiaeList = createCommentMinutiaeList(String.format(AUTO_GENERATED_COMMENT_WITH_REASON,
-                entityModule.getModuleName()));
-        imports = imports.add(getImportDeclarationNodeWithAutogeneratedComment(KEYWORD_BALLERINA,
-                BalSyntaxConstants.PERSIST_MODULE, commentMinutiaeList, null));
-        imports = imports.add(getImportDeclarationNode(BalSyntaxConstants.KEYWORD_BALLERINA,
-                KEYWORD_JBALLERINA_JAVA_PREFIX, null));
+    public static SyntaxTree generateDbClientSyntaxTree(Module entityModule) throws BalException {
```

Comment:
Better if we can have different classes for each provider. We have an abstract class/interface with a set of method definitions and implement changes based on the provider instead of adding different functions. This is not scalable.

Reply:
We may not have time to do it now but should do it soon

Reply:
Can you create an issue for this?

Reply:
Issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/4345